public/modules/custom/grants_handler/src/Plugin/WebformHandler/GrantsHandler.php

```diff
@@ -502,6 +502,37 @@ public function alterForm(array &$form, FormStateInterface $form_state, WebformS
         ->addError('Application period is closed, no further editing is allowed.');
       $form['#disabled'] = TRUE;
     }
+
+    $all_current_errors = $this->grantsFormNavigationHelper->getAllErrors($webform_submission);
+
+    // Loop through errors.
+    foreach ($all_current_errors as $pageName => $page) {
+      // Loop through errors in one page.
+      foreach ($page as $errorKey => $error) {
+        // Some errors are built like errorName][errorSelectValue.
+        // These variables separate the array keys in them.
+        $errorName = strtok($errorKey, ']');
+        $errorSelectValue = substr($errorKey, strpos($errorKey, '[') + 1);
+        if (isset($form['elements'][$pageName][$errorName])) {
+          $form['elements'][$pageName][$errorName]['#attributes']['class'][] = 'has-error';
+        }
+        else {
+          foreach ((array) $form['elements'][$pageName] as $fieldName => $element) {
+            if (!str_starts_with($fieldName, '#')) {
+              if (isset($form['elements'][$pageName][$fieldName][$errorName]['#webform_composite_elements'][$errorSelectValue])) {
+                $storage = $form_state->getStorage();
```

Comment:
This should be defined outside of foreach.

This way you'll be getting that storage as many times as the foreach is run.

Reply:
Storage variable I mean.